File: frontend/src/services/meta-reports.ts
import { apiFetch } from './api'

// ─── Types ───────────────────────────────────────────────

export type MetaReportLevel = 'campaign' | 'adset' | 'ad'

export interface MetaInsightParams {
  startDate: string
  endDate: string
  level?: MetaReportLevel
  adAccountId?: string
  limit?: number
}

export interface MetaInsightResponse {
  data: Record<string, unknown>[]
}

// ─── API ─────────────────────────────────────────────────

export function fetchMetaInsights(
  params: MetaInsightParams,
): Promise<MetaInsightResponse> {
  const sp = new URLSearchParams()
  sp.set('start_date', params.startDate)
  sp.set('end_date', params.endDate)
  if (params.level) sp.set('level', params.level)
  if (params.adAccountId) sp.set('ad_account_id', params.adAccountId)
  if (params.limit !== undefined) sp.set('limit', String(params.limit))
  return apiFetch<MetaInsightResponse>(`/api/meta/reports/insights?${sp.toString()}`)
}

File: backend/routes/meta_reports.py
from fastapi import APIRouter, Query
from meta_ads.api.reports import MetaReportService

router = APIRouter(prefix="/meta/reports", tags=["Meta 报表"])


@router.get("/insights")
async def account_insights(
    start_date: str = Query(...),
    end_date: str = Query(...),
    level: str = Query("campaign"),
    ad_account_id: str | None = Query(None),
    limit: int = Query(50),
):
    data = await MetaReportService(ad_account_id).get_account_insights(
        start_date=start_date, end_date=end_date, level=level, limit=limit,
    )
    return {"data": data.get("data", [])}
